### PR TITLE
Added refreshStorageFromDb and used it before printing identity

### PR DESCRIPTION
**Pull request by sengsational**
Issue #265

Description:
Earlier, I had observed that if the fields of the storage object were refreshed from the database before printing the identity PDF, then the verify block was of proper length, whereas if this read was not done, the error of the issue (longer than expected verify block) was printed.

I examined both `IdentityPrintDocumentAdapter` and also `ShowIdentityActivity`.  The latter activity appeared to already be refreshing the storage object from the database before use, so no changes appear to be required for that activity.  Those two places are where the problem `getVerifyingRecoveryBlock()` is being used in a new install / new identity situation that the issue describes.

Changes:
Added a method that refreshes the content of the storage object from the database and uses that method before printing the PDF.

### CODE REVIEW

**Comment by kalaspuffar** [comment]
I think I got the build system up and running again so a rebase could be good.

**Comment by kalaspuffar** [comment]
I will wait for answers on the review questions and before merging this pull request.

**Comment by sengsational** [comment]
There are no questions in the Mar 24, 2019, 9:39 PM EDT review...just wanted to make a note that my own testing was successful.

File: app/src/main/java/org/ea/sqrl/processors/SQRLStorage.java [hunk 1]
```diff
@@ -140,6 +143,18 @@ public boolean needsReload(byte[] identityData) {
...
 
+    public void refreshStorageFromDb(Activity activity) throws Exception {
+        SharedPreferences sharedPref = activity.getApplication().getSharedPreferences(
+                "org.ea.sqrl.preferences",
+                Context.MODE_PRIVATE
+        );
+        long currentId = sharedPref.getLong("current_id", 0);
+        IdentityDBHelper aDbHelper = new IdentityDBHelper(activity);
+        byte[] identityData = aDbHelper.getIdentityData(currentId);
+        SQRLStorage sqrlStorage = SQRLStorage.getInstance();
```
[the remark this answers is not in the document]
Comment (sengsational) [reply app/src/main/java/org/ea/sqrl/processors/SQRLStorage.java]: You're right.  It doesn't make sense to get instance.  Code was copied from elsewhere and I didn't think hard enough.  Needs to be moved to the Utils class anyway, as your earlier comment suggested.

File: app/src/main/java/org/ea/sqrl/processors/SQRLStorage.java [hunk 2]
```diff
@@ -140,6 +143,18 @@ public boolean needsReload(byte[] identityData) {
         return !Arrays.equals(identityData, oldData);
     }
 
+    public void refreshStorageFromDb(Activity activity) throws Exception {
```
[the remark this answers is not in the document]
Comment (sengsational) [reply app/src/main/java/org/ea/sqrl/processors/SQRLStorage.java]: You're right; code should be in the Utils class.

**Pull request merged**